actionview/lib/action_view/helpers/asset_url_helper.rb

```diff
@@ -118,16 +118,67 @@ module Helpers
     module AssetUrlHelper
       URI_REGEXP = %r{^[-a-z]+://|^(?:cid|data):|^//}i
 
-      # Computes the path to asset in public directory. If :type
-      # options is set, a file extension will be appended and scoped
-      # to the corresponding public directory.
+      # This is the entry point for all assets.
+      # When using the asset pipeline (i.e. sprockets and sprockets-rails), the
+      # behavior is "enhanced". You can bypass the asset pipeline by passing in
+      # <tt>public_folder: true</tt> to the options.
       #
       # All other asset *_path helpers delegate through this method.
       #
-      #   asset_path "application.js"                     # => /assets/application.js
-      #   asset_path "application", type: :javascript     # => /assets/application.js
-      #   asset_path "application", type: :stylesheet     # => /assets/application.css
-      #   asset_path "http://www.example.com/js/xmlhr.js" # => http://www.example.com/js/xmlhr.js
+      # === With the asset pipeline
+      #
+      # All options passed to +asset_path+ will be passed to +compute_asset_path+
+      # which is implemented by sprockets-rails.
+      #
+      #      asset_path("application.js") # => "/assets/application-60aa4fdc5cea14baf5400fba1abf4f2a46a5166bad4772b1effe341570f07de9.js"
+      #
+      # === Without the asset pipeline (<tt>public_folder: true</tt>)
+      #
+      # Accepts a <tt>type</tt> option that can specify the asset's extension. No error
+      # checking is done to verify the source passed into +asset_path+ is valid
+      # and that the file exists on disk.
```

Comment:
Not the biggest fan of `public_folder`. It implies that we're doing something special with this, when all we do is pass this directly through. I'd think something negative like `skip_pipeline: true` would be clearer for that. Would match this documentation as well.